CDP4Reporting/DataSource/ReportingDataSourceRow.cs

```diff
@@ -0,0 +1,52 @@
+﻿// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="ReportingDataSourceRow.cs" company="RHEA System S.A.">
+//    Copyright (c) 2015-2020 RHEA System S.A.
+//
+//    Author: Sam Gerené, Alex Vorobiev, Alexander van Delft, Cozmin Velciu, Adrian Chivu
+//
+//    This file is part of CDP4-IME Community Edition.
+//    The CDP4-IME Community Edition is the RHEA Concurrent Design Desktop Application and Excel Integration
+//    compliant with ECSS-E-TM-10-25 Annex A and Annex C.
+//
+//    The CDP4-IME Community Edition is free software; you can redistribute it and/or
+//    modify it under the terms of the GNU Affero General Public
+//    License as published by the Free Software Foundation; either
+//    version 3 of the License, or any later version.
+//
+//    The CDP4-IME Community Edition is distributed in the hope that it will be useful,
+//    but WITHOUT ANY WARRANTY; without even the implied warranty of
+//    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//    GNU Affero General Public License for more details.
+//
+//    You should have received a copy of the GNU Affero General Public License
+//    along with this program. If not, see <http://www.gnu.org/licenses/>.
+// </copyright>
+// --------------------------------------------------------------------------------------------------------------------
+
+namespace CDP4Reporting.DataSource
+{
+    using CDP4Common.EngineeringModelData;
+    using CDP4Common.SiteDirectoryData;
+
+    /// <summary>
+    /// Abstract base class from which all row representations for a <see cref="ReportingDataSourceClass{T}"/> need to derive.
+    /// </summary>
+    public abstract class ReportingDataSourceRow
+    {
+        /// <summary>
+        /// The associated <see cref="ElementBase"/>.
+        /// </summary>
+        protected internal ElementBase ElementBase { get; internal set; }
+
+        /// <summary>
+        /// Flag indicating whether the row matches the filtered criteria defined in <see cref="CategoryHierarchy"/>.
+        /// Note that when this is false, all values will be null on the row.
+        /// </summary>
+        public bool IsVisible { get; internal set; }
+
+        /// <summary>
+        /// The owner <see cref="DomainOfExpertise"/> of the associated <see cref="ElementBase"/>.
+        /// </summary>
+        protected DomainOfExpertise Owner => this.ElementBase.Owner;
```

Comment:
If we know what part of the owner `DomainOfExpertise` is needed to be used in the reports, we could change this to return the `Name` or `ShortName` directly (as public), thereby reducing the need to have a getter specifically declared in the user code as the automatic public getter to column conversion will pick it up.

I initially implemented it to return the full object because at the time we didn't know which of the fields would be used.

Reply:
The automatic getter is a smart idea, because the Name and ShortName will be added to the result datatable automatically.
But then the Name and ShortName will be added to the result datatable automatically!!!! ;)
Don't think we always want that to happen.

Also, at the moment IsVisible is always visible in the result datatable, thus visible in the report designer, and that is something we probably don't want. Its value is always true. 
I'll remove that one too and also rename Owner to ElementBaseOwner.